cas-server-support-saml/src/main/java/org/jasig/cas/support/saml/SamlServletContextListener.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.support.saml;
+
+import org.jasig.cas.support.saml.authentication.principal.SamlService;
+import org.jasig.cas.util.UniqueTicketIdGenerator;
+import org.jasig.cas.web.support.ArgumentExtractor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.ServletRegistration;
+import javax.servlet.annotation.WebListener;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Initializes the CAS root servlet context to make sure
+ * SAML validation endpoint can be activated by the main CAS servlet.
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+@WebListener
+@Component
+public class SamlServletContextListener implements ServletContextListener, ApplicationContextAware {
```

Comment:
Correct, they cannot. One deals with the root web application context, and one deals with the servlet application context which is a child of the former.